core/src/main/java/hudson/util/io/RewindableFileOutputStream.java

```diff
@@ -51,8 +54,8 @@ private synchronized OutputStream current() throws IOException {
             if (!closed) {
                 FileUtils.forceMkdir(out.getParentFile());
                 try {
-                    current = new FileOutputStream(out,false);
-                } catch (FileNotFoundException e) {
+                    current = Files.newOutputStream(out.toPath(), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+                } catch (FileNotFoundException | NoSuchFileException e) {
```

Comment:
Uh oh, you are right. http://stackoverflow.com/questions/14263725/java-files-write-nosuchfileexception. Does it make sense to catch it in other usages then?